src/library/coverart.cpp

```diff
@@ -28,20 +31,69 @@ QString typeToString(CoverInfo::Type type) {
     return "INVALID TYPE VALUE";
 }
 
+QString coverInfoRelativeToString(const CoverInfoRelative& infoRelative) {
+    return typeToString(infoRelative.type) % QLatin1Literal(",") %
+           sourceToString(infoRelative.source) % QLatin1Literal(",") %
+           infoRelative.coverLocation % QLatin1Literal(",") %
+           QLatin1Literal("0x") % QString::number(infoRelative.hash, 16);
+}
+
+QString coverInfoToString(const CoverInfo& info) {
+    return coverInfoRelativeToString(info) % QLatin1Literal(",") %
+           info.trackLocation % QLatin1Literal(",");
+}
+} // anonymous namespace
+
+
+bool operator==(const CoverInfoRelative& a, const CoverInfoRelative& b) {
+    return a.source == b.source &&
+            a.type == b.type &&
+            a.coverLocation == b.coverLocation &&
+            a.hash == b.hash;
+}
+
+bool operator!=(const CoverInfoRelative& a, const CoverInfoRelative& b) {
+    return !(a == b);
+}
+
+QDebug operator<<(QDebug dbg, const CoverInfoRelative& infoRelative) {
+    return dbg.maybeSpace() << QString("CoverInfoRelative(%1)")
+            .arg(coverInfoRelativeToString(infoRelative));
+}
+
+bool operator==(const CoverInfo& a, const CoverInfo& b) {
+    return static_cast<CoverInfoRelative>(a) ==
```

Comment:
Doesn't this cast produce a temporary copy of a/b by invoking the copy constructor of CoverInfoRelative with a/b as the argument? I would cast to const-ref, i.e. "const CoverInfoRelative&"